SCons/Environment.py

```diff
@@ -2407,10 +2411,15 @@ def __contains__(self, key):
             return 1
         return self.__dict__['__subject'].__contains__(key)
 
-    def Dictionary(self):
+    def Dictionary(self, *args):
         d = self.__dict__['__subject'].Dictionary().copy()
         d.update(self.__dict__['overrides'])
-        return d
+        if not args:
```

Comment:
Why change Dictionary()'s implementation in this PR?

Reply:
oh I c.. did you add this, but then didn't update the PR's title?

Reply:
wasn't in the title no: digging in here exposed a "neighbor" in OverrideEnvironment that was incomplete (because I used Dictionary in the first pass, and it failed, because it didn't implement passing args to Dictionary)